documentation/chatmeleon-swagger.yaml

```diff
@@ -283,6 +284,208 @@ paths:
+      security:
+        - bearerAuth: []
+      responses:
+        "200":
+          description: Relationship deleted successfully.
+        "401":
+          description: Unauthorized - JWT token is missing or invalid.
+        "403":
+          description: Unauthorized - User is unauthorized.
+
+      requestBody:
+        description: The ID of the other user to delete relationship.
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                relatedUserId:
+                  type: string
+                  pattern: "^[0-9a-f]{24}$"
+
+  /friends/search:
+    get:
+      description: Retrieve a page of friends associated with the current logged in user, with the support of cursor-based pagination ordered/based on a searched string.
+      tags:
+        - relationships
+      summary: Get friends of the current user.
+      parameters:
+        - name: name
+          in: query
+          description: The string of name that is being searched for. When this query is empty, it should return a page of friends in alphabetical order with cursor-based pagination.
+          schema:
+            type: string
+        - name: pageSize
+          in: query
+          description: The number of user one page. When this query is empty, a default value of 10 will be used. This only accepts positive values for pageSize.
+          schema:
+            type: integer
+            default: 10
+        - name: cursor
+          in: query
+          description: The ID of the most relevant user of the last messages page. When this query is empty, a default value of null will be used.
+          schema:
+            type: string
+            pattern: "^[0-9a-f]{24}$"
+            default: null
+      security:
+        - bearerAuth: []
+      responses:
+        "200":
+          description: An array of friends related to the searched name.
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  type: object
+                  properties:
+                    userId:
+                      type: string
+                      pattern: "^[0-9a-f]{24}$"
+                    name:
+                      type: string
+                    image:
+                      type: string
+                    relationshipType:
+                      type: string
+                      enum:
+                        - FRIEND
+                        - BLOCK
+                        - FOLLOWING
+                        - PENDING_FRIEND_REQUEST
+                        - SENDING_FRIEND_REQUEST
+                    originalConversationId:
+                      type: string
+                      pattern: "^[0-9a-f]{24}$"
+        "401":
+          description: Unauthorized - JWT token is missing or invalid.
+        "403":
+          description: Unauthorized - User is unauthorized.
+
+  /users/search:
```

Comment:
Same comments as `GET /friends/search`. But maybe we can combine these 2 search routes into one? And add a filter option like `/user/search?filter=friends` or something.

Reply:
true tho, @KhoaNgo18 consider this maybe

Reply:
vay chac bo cai social, lam chung voi cai route friend xong co cai filter=friend thoi